zbuild.h

```diff
@@ -85,7 +85,11 @@
 #  define PREFIX3(x) z_ ## x
 #  define PREFIX4(x) x ## 64
 #  define zVersion zlibVersion
-#  define z_size_t unsigned long
+#  if defined(_WIN64)
```

Comment:
Because using `size_t` or `uintptr_t` would require including a header in `zbuild.h` and that would introduce problems with header ordering in any file that includes `zbuild.h`. I really don't want to include all standard headers in `zbuild.h` just to avoid including them in wrong order in source files. `zbuild.h` is supposed to always be the first header included, so it can't include any headers that might have circular dependency, like `stdint.h` or `stdarg.h`.

Reply:
`zbuild.h` already includes `stdint.h`?

Reply:
Including `stdint.h` itself is not problem, the problem comes if also `stdarg.h` is included. Also, any header whose contents are not used will be filtered out by preprocessor. We use compiler's internal types intentionally to catch any "bugs" in compiler configuration. In the past it was common for files to be compiled differently inside Visual Studio IDE and command-line using nmake Makefiles. If the types don't match with the ones used in `zlib.h` or `zllb-ng.h`, the error is caught instead of getting masked out.